### PR TITLE
Authenticate with GCE 

### LINKED ISSUE #6: Authenticate with GCE
Add UI elements to allow Google Cloud authentication tokens to be captured. This requires https://github.com/galaxyproject/cloudlaunch/issues/78

Technically, GCE requires a credentials file or a dict (see https://github.com/gvlproject/cloudbridge/blob/gce/cloudbridge/cloud/providers/gce/provider.py#L141), which contains a number of fields. Rather than exposing each field, it'll probably be better to just store the contents of the entire file as a blob and use it as a credentials dict when authenticating.

### PR DESCRIPTION
**Pull request by machristie**
This fixes #6.

### CODE REVIEW

**Comment by afgane** [comment]
This looks very nice!
It's too bad dome of the CloudBridge GCE functionality is still missing to test all of this.

A few times, I've run in the the case where I get UI issue like this:
image
Anyone else seen it? Refreshing the page fixes it but it appeared more than once on the profile page as well as application page. Here's the output from the console:
image


Thanks for also fixing the cloud-based filtering when adding credentials.

**Comment by machristie** [comment]
That looks related to a change I made. I'm not getting that particular issue but I'll see if I can fix it.

**Comment by machristie** [comment]
@afgane I've committed a change that may fix the problem, but I'm not sure. The fix is to use `cloud.name` for the displayed name of the cloud when the cloud can't be changed.  The `cloud.text` property is only set by ng2SelectAdjust for adding that property which is needed by ng2select.

**Comment by afgane** [comment]
Looks like we'll have to dig more. It seemed better as it didn't happen for a number of clicks but switching between a couple of clouds back and forth triggered it. It might just be the ng2select component too that's buggy because now the version dropdown locked up as well:

image

**Comment by machristie** [comment]
I've pushed a commit to the gce branch that should fix some of the problem that Enis mentions above. The fix is related to the cloud type filtering I added. There's still another issue with the UbuntuConfigComponent not implementing some methods that I'm still looking into.

**Comment by machristie** [comment]
TODO:
- [ ] When editing existing credential, don't allow changing the cloud, in cloud-credentials-editor

**Comment by machristie** [comment]
Looks like there are conflicts again so I'll rebase again.

**Comment by machristie** [comment]
@afgane @nuwang I've finished rebasing and testing. This PR is ready to be reviewed now.

File: src/app/my-profile/components/cloud-credentials-editor/gce-cred-editor/gce-cred-editor.component.html [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+        <!-- Credentials -->
+        <div class="form-group is-empty"
+         [ngClass]="{'has-error': '!credentials.valid'}">
+            <label for="id_credentials" class="control-label">GCE Credentials</label>
+            <textarea id="id_credentials" name="credentials" class="form-control input-md" formControlName="credentials" rows="12"></textarea>
+            <span class="help-block"> Copy your GCE credentials json file into this field, or use the upload button above.</span>
+            <span class="material-input"></span>
+        </div>
+    </div>
+</div>
```
Comment (nuwang) [inline src/app/my-profile/components/cloud-credentials-editor/gce-cred-editor/gce-cred-editor.component.html]: This html was changed to use angular-material  in the other providers: https://github.com/galaxyproject/cloudlaunch-ui/blob/master/src/app/my-profile/components/cloud-credentials-editor/aws-cred-editor/aws-cred-editor.component.html
Would you mind copying it over from there or I can do it if you're short on time.

**Pull request merged**